docs/tutorial.rst

```diff
@@ -0,0 +1,987 @@
+    Connection: keep-alive
+    Content-Length: 30
+    Content-Type: application/json
+    Host: kinto.dev.mozaws.net
+    User-Agent: HTTPie/0.9.2
+
+    {
+        "data": {
+            "status": "doing"
+        }
+    }
+
+    HTTP/1.1 200 OK
+    Access-Control-Expose-Headers: Backoff, Retry-After, Alert
+    Backoff: 10
+    Connection: keep-alive
+    Content-Length: 254
+    Content-Type: application/json; charset=UTF-8
+    Date: Thu, 18 Jun 2015 15:50:03 GMT
+    Server: nginx/1.4.6 (Ubuntu)
+
+    {
+        "data": {
+            "description": "Write a tutorial explaining Kinto", 
+            "id": "23eed462-c063-4ae0-81b0-8bf2210bfe86", 
+            "last_modified": 1434642603605, 
+            "status": "doing"
+        }, 
+        "permissions": {
+            "write": [
+                "basicauth:10ea4e5fbf849196a4fe8a9c250b737dd5ef17abbeb8f99692d62828465a9823"
+            ]
+        }
+    }
+
+There you should ask yourself, what happens if another device already
+updated the record in between, will I override its change?
+
+You've got two conflicts resolution behaviors:
+
+- Server wins, in that case the server will reject changes in case
+  something changed on server side.
+- Client wins, in that case the change will override previous changes
+
+The previous call is the Client wins behavior.
+
+In case you want the server to prevent you from overridding changes,
+you must send the ``If-Match`` header:
+
+Let say, we didn't refresh the server since our first POST and we send
+the ETag we had back then ``"1434641515332"``:
+
+.. code-block:: http
+
+    $ echo '{"data": {"status": "doing"}}' | \
+        http PATCH https://kinto.dev.mozaws.net/v1/buckets/default/collections/tasks/records/23eed462-c063-4ae0-81b0-8bf2210bfe86 \
+            If-Match:'"1434641515332"' \
+            -v  --auth 'user:password'
+
+    PATCH /v1/buckets/default/collections/tasks/records/23eed462-c063-4ae0-81b0-8bf2210bfe86 HTTP/1.1
+    Accept: application/json
+    Accept-Encoding: gzip, deflate
+    Authorization: Basic dXNlcjpwYXNzd29yZA==
+    Connection: keep-alive
+    Content-Length: 29
+    Content-Type: application/json
+    Host: kinto.dev.mozaws.net
+    If-Match: "1434641515332"
+    User-Agent: HTTPie/0.9.2
+
+    {
+        "data": {
+            "status": "done"
+        }
+    }
+
+    HTTP/1.1 412 Precondition Failed
+    Access-Control-Expose-Headers: Backoff, Retry-After, Alert
+    Connection: keep-alive
+    Content-Length: 98
+    Content-Type: application/json; charset=UTF-8
+    Date: Thu, 18 Jun 2015 16:08:31 GMT
+    ETag: "1434642603605"
+    Last-Modified: Thu, 18 Jun 2015 15:50:03 GMT
+    Server: nginx/1.4.6 (Ubuntu)
+
+    {
+        "code": 412, 
+        "errno": 114, 
+        "error": "Precondition Failed", 
+        "message": "Resource was modified meanwhile"
+    }
+
+The server reject the modification with a 412 error code.
+
+In order to fix that, we can either ask for the record we tried to
+update:
```

Comment:
In order to merge both version, we can fetch the last version of this single record